scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,100 @@ def visit_functiondef(self, node):
     # this line makes it work for async functions
     visit_asyncfunctiondef = visit_functiondef
 
+class CheckEnum(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-enum"
+    priority = -1
+    msgs = {
+        "C4746": (
+            "The enum must use uppercase naming. "
+            "https://azure.github.io/azure-sdk/python_design.html#enumerations",
+            "enum-must-be-uppercase",
+            "Capitalize enum name.",
+        ),
+        "C4747":(
+            "The enum must inherit from CaseInsensitiveEnumMeta. "
+            "https://azure.github.io/azure-sdk/python_implementation.html#extensible-enumerations",
+            "enum-must-inherit-case-insensitive-enum-meta",
+            "Inherit CaseInsensitiveEnumMeta.",
+        ),
+    }
+    options = (
+        (
+            "ignore-enum-must-be-uppercase",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not be capitalized.",
+            },
+        ),
+        (
+            "ignore-enum-must-inherit-case-insensitive-enum-meta",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not inherit CaseInsensitiveEnumMeta.",
+            },
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckEnum, self).__init__(linter)
+
+    def visit_classdef(self, node):
+        """Visits every enum class.
+
+        :param node: ast.ClassDef
+        :return: None
+        """
+        try:
+            enum_class = False
+            case_insensitive_meta = False
+            
+            #Python3 format declares CaseInsensitiveEnumMeta as a metaclass in enum classes
+            if node.declared_metaclass():
+                if node.declared_metaclass().name:
+                    if node.declared_metaclass().name == "CaseInsensitiveEnumMeta":   # or node.declared_metaclass().name == "_CaseInsensitiveEnumMeta":
```

Comment:
Does the commented out portion of this line need to be removed? Or included?

Reply:
Removed!